src/runtime_src/core/pcie/common/dmatest.h

```diff
@@ -144,7 +148,9 @@ namespace xcldev {
         }
 
         ~DMARunner()
-        {}
+        {
+	    mBOList.clear();
```

Comment:
Why?

Reply:
The BOs are getting freed after hw destructor getting called. 
Hence, after hw context destructed there are some BOs are still present.

Reply:
But mBOList is an automatic object, it is destructed when DMARunner goes out of scope.  That's what I meant by "Why?"

Reply:
Buffers are keeping an information about hw context. If buffers are freed before hw context, then we are losing track. We need to make sure all buffers are freed up before hardware context goes out of scope.

Reply:
Ok, got it, but order of destruction is opposite order of construction so it is implicitly correct anyway.  But I like being explicit about it, but that definitely requires a comment in the code before calling clear(), it is not at all obvious.

Reply:
Added a comments for the same